eng/common/scripts/X509Certificate2/README.md

```diff
@@ -0,0 +1,24 @@
+Powershell module for generating self-signed x509 certificates
+
+Usage:
+
+```powershell
+Import-Module -Name eng/scripts/X509Certificate2.psm1 # assumes $PWD is repo root
```

Comment:
Does this have to be a module? In general we have moved away from modules do to loading issues and just use a script with helper functions.

Reply:
It was developed that way, but I'm also happy to spend a few minutes turning it into a cmdlet instead.

Reply:
A cmdlet is a module right? At any rate I'm OK with this for now and if we hit any issues we can reevaluate.

Reply:
I think of top-level scripts as cmdlets

Reply:
Modules contain 1 or more cmdlets (or none, actually; could just be type or view data). Scripts can be cmdlets. Modules can be unloaded, and because these are all a bunch of related cmdlets I didn't want a bunch of one-off scripts that are harder to keep in sync. In my experience, I've never had problems with modules. They solved the fundamental problems with snapins.

Reply:
We have definitely had issues with module loading and unloading, especially during development of them as you often times need to restart your shell to unload successfully. We also have issues when loading modules in parallel when running steps which causes race conditions and other interesting things. That is why we moved to having a set of shared helper scripts/functions instead of modules. That said if I don't have concerns with keeping this as a module if it is already working well as one.